### PR TITLE
fix: Comet should not translate try_sum to native sum expression

### LINKED ISSUE #276: Comet should not translate try_sum to native sum expression
### Describe the bug

One more test failure found while working in #250.

SQLQuerySuite:

```
[info] - SPARK-38548: try_sum should return null if overflow happens before merging *** FAILED *** (179 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 2306.0 failed 1 times, most recent failure: Lost task 1.0 in stage 2306.0 (TID 2699) (b8cdd0fa496d executor driver): java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
[info] 	at org.apache.spark.sql.vectorized.ColumnarBatchRow.getBoolean(ColumnarBatchRow.java:97)
```

It is because the query uses `try_sum` expression. Currently Comet ignores `evalMode` parameter for `sum` and `average`. Because they might have different aggregation buffer or behavior, we should not simply translate `try_sum` to `sum`.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #276.

### CODE REVIEW

**Comment by viirya** [comment]
cc @huaxingao @sunchao @andygrove

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala
Comment (viirya) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:207]: Hmm, this is introduced in Spark 3.4. So Spark 3.2/3.3 builds are failed due to this change. I need to add a shim layer...

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala
Comment (viirya) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:205]: Yea, we should do it. We probably need to implement native try_sum expression because DataFusion might not have it.

File: spark/src/main/scala/org/apache/comet/shims/ShimQueryPlanSerde.scala
Comment (viirya) [reply spark/src/main/scala/org/apache/comet/shims/ShimQueryPlanSerde.scala]: Okay

**Review by huaxingao — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by huaxingao** [comment]
There are some format issues that need to be fixed.

**Comment by viirya** [comment]
I'd need one committer approval to merge this. 🙂

**Review by sunchao — APPROVED** [review]

**Comment by viirya** [comment]
Thank you @sunchao 😄

**Comment by viirya** [comment]
Merged. Thanks @huaxingao @sunchao @advancedxy

**Pull request merged**
